crates/ruff_linter/src/rules/pycodestyle/overlong.rs

```diff
@@ -125,8 +125,8 @@ impl<'a> StrippedLine<'a> {
         let comment_range = comment_range - line.start();
         let comment = &line.as_str()[comment_range];
 
-        // Ex) `# type: ignore`
-        if is_pragma_comment(comment) {
+        // Ex) `# type: ignore` or `# test # noqa: RUF100`
+        if is_pragma_comment(comment) || contains_pragma_after_hash(comment) {
```

Comment:
I don't think this fix is correct and it shows in the ecosystem results. 

The idea is to only count the characters up to, but not including, the first pragma comment that we see in the line length computation.